### PR TITLE
fixup generate task after move to hugo

### PR DESCRIPTION
**Pull request by stevesloka**
Signed-off-by: Steve Sloka <slokas@vmware.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -198,7 +198,7 @@ lint-flags:
 
 .PHONY: generate
 generate: ## Re-generate generated code and documentation
-generate: generate-rbac generate-crd-deepcopy generate-crd-yaml generate-deployment generate-api-docs generate-metrics-docs generate-uml
```
Comment (youngnick) [inline Makefile:201]: After this change, how do we update the metrics docs? They're still present on the "Prometheus metrics" page, but it seems like now we won't be able to manage them programmatically? If the plan is to fix them up later, that's fine, but a TODO here or an issue link would be great.

**Review by youngnick — COMMENTED** [review]
LGTM in general, but would like to know what the plan is for metrics documentation.

**Comment by sunjayBhatia** [comment]
Just submitted a couple PRs to fix other targets/scripts that are part of the `codegen` CI job:
- https://github.com/projectcontour/contour/pull/3716
- https://github.com/projectcontour/contour/pull/3715

File: site/_data/refdocs/config.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (sunjayBhatia) [inline site/_data/refdocs/config.json:1]: should this content be added? the hugo site format doesnt have a `_data` dir so this seems entirely new

File: Makefile (hunk 1 above)
Comment (stevesloka) [reply Makefile:201]: Why do we need to dynamically generate that page? It rarely changes and is just overhead to me. Additionally, the bits to generate the docs needed to be added by hand.

File: site/_data/refdocs/config.json (hunk 2 above)
Comment (stevesloka) [reply site/_data/refdocs/config.json:1]: It's used for the generator to build the site. The _data dir doesn't get published over to the hugo site from my tests.

Comment (sunjayBhatia) [reply site/_data/refdocs/config.json:1]: from the generate-api-docs.sh script looks like these templates shouldnt have been deleted, maybe we can modify that script to look for them somewhere else to keep with the hugo structure?

File: Makefile (hunk 1 above)
Comment (sunjayBhatia) [reply Makefile:201]: removing it also means we don't run promlint on our metrics, which could mean we add something not ideal as well

Comment (stevesloka) [reply Makefile:201]: https://github.com/projectcontour/contour/issues/3717

**Review by sunjayBhatia — APPROVED** [review]

**Review by danehans — APPROVED** [review]

**Pull request merged**